### PR TITLE
Specify form selector

### PR DESCRIPTION
**Pull request by julianrubisch**
# Enhancement

## Description

This PR adds the possibility to declare a custom form selector to be used when scooping up form parameters:

```html
<div data-reflex="..." data-reflex-form-selector="#custom_form">....</div>
```

```js
this.stimulate("MyReflex#action", this.element, { formSelector: "#custom_form" })
```

**Note:** This is based on my work on #517, the actual diff is much smaller: https://github.com/stimulusreflex/stimulus_reflex/commit/4c39aff63f7b902a684e9becbc4672595cff32e2

## Why should this be added

At the moment I'm building a `WizardReflex` where the actual form is separated from the navigation buttons/tabs, thus form params wouldn't be submitted.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] Checks (StandardRB & Prettier-Standard) are passing
- [X] This is not a documentation update

Please note that the best way to suggest changes or updates to the [documentation](https://docs.stimulusreflex.com) is to [join Discord](https://discord.gg/stimulus-reflex) and leave a note in the #docs channel. Any documentation updates posted as PRs cannot be accepted at this time. :heart:

### CODE REVIEW

**Comment by leastbad** [comment]
Out of the gate: there are documentation changes in here. Please, please take them out.

**Comment by julianrubisch** [comment]
easy, this is going to rebased once #517 lands

**Comment by leastbad** [comment]
I can't stress enough how poorly GitBook handles non-clean merges. I get super jumpy when I see things creep in...

File: javascript/stimulus_reflex.js
Comment (marcoroth) [inline javascript/stimulus_reflex.js:194]:
```suggestion
        const form = reflexElement.closest(reflexData.formSelector || 'form')
```

I think we should still use `Element.closest()` to determine the form since there could be more than one form. Also since it's not mandatory to pass in an ID.

**Review by marcoroth — COMMENTED** [review]
I like the general direction of this PR 👍🏼

**Comment by julianrubisch** [comment]
> I can't stress enough how poorly GitBook handles non-clean merges. I get super jumpy when I see things creep in...

Admittedly, it's hard to reason about this way. Still I wanted to put it out there. We'll have to wait for @hopsoft to review #517

**Comment by leastbad** [comment]
Essentially, every paragraph in the docs that has at least one changed character gets duplicated 2-3 times, showing a before/after... but it's all just text. You have to scan every page looking for dupes, and fussing to make sure the right version persists.

It takes hours.

File: javascript/stimulus_reflex.js
Comment (julianrubisch) [reply javascript/stimulus_reflex.js]: hmm in my case this didn't work because it's not a child, but a sibling... reverting...

**Review by hopsoft — APPROVED** [review]
Nice work. Let's wait for #517 to land then get this merged.

File: javascript/stimulus_reflex.js
Comment (marcoroth) [reply javascript/stimulus_reflex.js:194]: But then I would recommend to do this instead:

```suggestion
        const form =
          reflexElement.closest(reflexData.formSelector) ||
          document.querySelector(reflexData.formSelector) ||
          reflexElement.closest('form')
```

**Pull request merged**
